dev/io.openliberty.jakartaee10.internal_fat/fat/src/io/openliberty/jakartaee10/internal/tests/EE10FeatureCompatibilityTest.java

```diff
@@ -436,7 +388,11 @@ private void checkFeatures(String baseFeature, Queue<String> featureQueue, Map<S
         while ((feature = featureQueue.poll()) != null) {
             Log.info(c, "checkFeatures", "start testing: " + feature);
             featuresToTest.set(1, feature);
-            boolean expectToConflict = nonEE10JavaEEFeatures.contains(feature) || nonEE10MicroProfileFeatures.contains(feature) || incompatibleValueAddFeatures.contains(feature);
+            if (!compatibleFeatures.contains(feature) && !incompatibleFeatures.contains(feature)) {
+                // Don't test this feature
+                continue;
+            }
```

Comment:
Won't this cause us to possibly miss features.  There was a reason I didn't have this logic.

Reply:
Yes, and I can see why it was written like that but I want to change it for two reasons:

1. During development when we're trying to make features compatible with EE10, it's useful to be able to say "I don't want to assert compatibility of this feature either way yet. If I can't do this then as I change my feature to make it compatible with EE10, I not only need to get my own FAT tests repeating and passing, I also need to change this test to expect the change I'm making (and any other knock-on changes, maybe making my feature compatible has also made another feature compatible that I wasn't aware of). In practise, that means I need to run this test, see what fails and change the test to expect the actual result.

2. With the way that `getIncompatibleFeatures()` is written, it's very easy to see which features aren't included in either set. The starting point is that everything not compatible is incompatible, and then we take some features out that we don't want to assert incompatibility for. As we finish development on MP6, it should be easy to verify that we're not missing any features.